python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -357,35 +357,35 @@ def _create_node(self, node_config, tags, count):
         # single SubnetId before invoking the AWS API.
         subnet_ids = conf.pop("SubnetIds")
 
+        # update config with min/max node counts and tag specs
+        conf.update({
+            "MinCount": 1,
+            "MaxCount": count,
+            "TagSpecifications": tag_specs
+        })
+
+        cli_logger_tags = {}
         for attempt in range(1, BOTO_CREATE_MAX_RETRIES + 1):
             try:
-                subnet_id = subnet_ids[self.subnet_idx % len(subnet_ids)]
-
-                self.subnet_idx += 1
-                conf.update({
-                    "MinCount": 1,
-                    "MaxCount": count,
-                    "NetworkInterfaces": [{
-                        "AssociatePublicIpAddress": True,
-                        "DeviceIndex": 0,
-                        "SubnetId": subnet_id,
-                    }],
-                    "TagSpecifications": tag_specs
-                })
-                _ = conf.pop("SubnetId", None)
-                security_group_ids = conf.pop("SecurityGroupIds", None)
-                if security_group_ids is not None:
-                    conf["NetworkInterfaces"][0]["Groups"] = security_group_ids
+                if "NetworkInterfaces" in conf:
+                    net_ifs = conf["NetworkInterfaces"]
+                    # remove security group IDs previously copied from network
+                    # interfaces (create_instances call fails otherwise)
+                    conf.pop("SecurityGroupIds", None)
```

Comment:
You need to pop SubnetId as well!

Reply:
SubnetId is taken care of by the conflict logic back up here:
https://github.com/ray-project/ray/blob/d8013282173a9b41d942fd600a06bb48262ddbd0/python/ray/autoscaler/_private/aws/config.py#L784


I guess it wouldn't hurt to be defensive by popping again.

Reply:
To safeguard this further, the unit test I've just added will fail if SubnetId ever manages to sneak in, since it won't match the expected input params stubbed out for `run_instances`.